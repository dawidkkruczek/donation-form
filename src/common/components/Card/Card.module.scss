.root {
  background: #fff;
  box-shadow: 0px 16px 32px rgba(30, 42, 50, 0.08);

  @media screen and (min-width: 768px) {
    border-radius: 5px;
  }
}

.content {
  padding: 32px 24px;

  @media screen and (min-width: 768px) {
    padding: 32px 40px;
  }
}
