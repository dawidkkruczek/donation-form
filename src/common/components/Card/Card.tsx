import React, { PropsWithChildren, ReactElement } from 'react';
import classNames from 'classnames';
import styles from './Card.module.scss';
import Header, { Props as HeaderProps } from './Header/Header';

export interface Props extends PropsWithChildren {
  className?: string;
  headerProps?: HeaderProps;
}

function Card({ children, className, headerProps }: Props): ReactElement {
  return (
    <div className={classNames(styles.root, className)}>
      {headerProps && <Header {...headerProps} />}
      <div className={styles.content}>{children}</div>
    </div>
  );
}

export default Card;
