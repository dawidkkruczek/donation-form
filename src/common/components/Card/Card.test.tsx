import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

describe('common/components/Card', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Card />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with header', () => {
    const { asFragment } = render(
      <Card
        headerProps={{
          icon: () => <svg />,
          onClose: () => {},
          subtitle: 'Subtitle',
          title: 'Title',
        }}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
