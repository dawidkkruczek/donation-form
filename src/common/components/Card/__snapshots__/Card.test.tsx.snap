// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/components/Card should render correctly 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="content"
    />
  </div>
</DocumentFragment>
`;

exports[`common/components/Card should render correctly with header 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="root"
    >
      <div
        class="content"
      >
        <svg />
        <h2
          class="title"
        >
          Title
        </h2>
        <div
          class="subtitle"
        >
          Subtitle
        </div>
      </div>
      <button
        aria-label="Close"
        class="close"
        type="button"
      >
        <svg
          class="closeIcon"
        >
          close-icon.svg
        </svg>
      </button>
    </div>
    <div
      class="content"
    />
  </div>
</DocumentFragment>
`;
